boardgamegeek/api.py

```diff
@@ -888,17 +888,23 @@ def game(self, name=None, game_id=None, choose=BGGChoose.FIRST, versions=False,
         while added_items and len(game.comments) < total:
             page += 1
 
+            params['page'] = page
             xml_root = request_and_parse_xml(self.requests_session,
                                              self._thing_api_url,
-                                             params={"id": game_id,
-                                                     "pagesize": 100,
-                                                     "comments": 1,
-                                                     "page": page})
+                                             params=params,
+                                             timeout=self._timeout,
+                                             retries=self._retries,
+                                             retry_delay=self._retry_delay)
+
+            xml_root = xml_root.find("item")
+            if xml_root is None:
+                msg = "invalid data for game id: {}{}".format(game_id, "" if name is None else " ({})".format(name))
+                raise BGGApiError(msg)
 
-            added_items = add_game_comments_from_xml(game, xml_root)
+            added_items, total = add_game_comments_from_xml(game, xml_root)
 
             try:
-                call_progress_cb(progress, len(game), game.comments)
+                call_progress_cb(progress, len(game.comments), total)
```

Comment:
The earlier version of this line raised an exception (`TypeError: object of type 'BoardGame' has no len()`). While I don't totally understand what this line does, I changed it to match the similar line earlier in this function.

Reply:
Right again. Makes me wonder how many people actually fetched comments 😆 

`call_progress_cb` is needed if the user wants to have some sort of progress feedback in his app, when fetching multiple pages of data (which can take a while). It will call the user provided `progress` function with two args, number of items and total, so he can compute an estimated completion percentage.